static/app/components/charts/groupStatusChart.tsx

```diff
@@ -0,0 +1,150 @@
+import {useMemo} from 'react';
+import styled from '@emotion/styled';
+
+import MarkLine from 'sentry/components/charts/components/markLine';
+import MiniBarChart from 'sentry/components/charts/miniBarChart';
+import {LazyRender} from 'sentry/components/lazyRender';
+import {t} from 'sentry/locale';
+import type {Group, TimeseriesValue} from 'sentry/types';
+import type {Series} from 'sentry/types/echarts';
+import {formatAbbreviatedNumber} from 'sentry/utils/formatters';
+import theme from 'sentry/utils/theme';
+
+function asChartPoint(point: [number, number]): {name: number | string; value: number} {
+  return {
+    name: point[0] * 1000,
+    value: point[1],
+  };
+}
+
+const EMPTY_STATS: ReadonlyArray<TimeseriesValue> = [];
+
+type Props = {
+  data: Group;
+  statsPeriod: string;
+  groupStatus?: string;
+  height?: number;
+  showMarkLine?: boolean;
+  showSecondaryPoints?: boolean;
+};
+
+function GroupStatusChart({
+  statsPeriod,
+  data,
+  groupStatus,
+  height = 24,
+  showMarkLine = false,
+  showSecondaryPoints = false,
+}: Props) {
+  const stats: ReadonlyArray<TimeseriesValue> = statsPeriod
+    ? data.filtered
+      ? data.filtered.stats?.[statsPeriod]
+      : data.stats?.[statsPeriod]
+    : EMPTY_STATS;
+
+  const secondaryStats: TimeseriesValue[] | null =
+    statsPeriod && data.filtered ? data.stats[statsPeriod] : null;
+
+  const [series, colors, emphasisColors]: [
+    Series[],
+    [string] | undefined,
+    [string] | undefined,
+  ] = useMemo(() => {
+    if (!stats || !stats.length) {
+      return [[], undefined, undefined];
+    }
+
+    let max = 0;
+    const marklinePoints: number[] = [];
+
+    for (let i = 0; i < stats.length; i++) {
+      const point = stats[i];
+      if (point[1] > max) {
+        max = point[1];
+      }
+      marklinePoints.push(point[1]);
+    }
+
+    const formattedMarkLine = formatAbbreviatedNumber(max);
+
+    const chartSeries: Series[] = [];
+    let chartColors: [string] | undefined = undefined;
+    let chartEmphasisColors: [string] | undefined = undefined;
+
+    if (showSecondaryPoints && secondaryStats && secondaryStats.length) {
+      chartSeries.push({
```

Comment:
Instead of the `let` above and pushing, you could just return the value in each of the if branches:

```ts
const chartSeries = [...]
return [chartSeries, chartColors, chartEmphasisColors]
```